molecule/test/unit/test_logger.py

```diff
@@ -177,3 +179,16 @@ def test_markup_detection_tty_no(mocker):
     assert not logger.should_do_markup()
     mocker.resetall()
     mocker.stopall()
+
+def test_logger_class(capsys, monkeypatch):
+
+    class FooLogger(logging.getLoggerClass()):
+        log = logging.getLogger('FooLogger')
+        pass
+
+    logging.setLoggerClass(FooLogger)
+
+    try:
+        log = FooLogger('foo')
+    except RecursionError as exc:
+        raise pytest.fail("CustomLogger caused recursion error: {0}".format(exc))
```

Comment:
Why are you catching this, and not just letting the error pass out to pytest automatically?

Reply:
not sure, I guess that also works :) ill change